src/ngraph/runtime/inference_engine/ie_executable.hpp

```diff
@@ -0,0 +1,50 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <ie_core.hpp>
+#include "ngraph/runtime/tensor.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace inference_engine
+        {
+            // A Inference Engine executable object produced by compiling an nGraph function.
+            class IE_Executable final : public Executable
+            {
+            public:
+                IE_Executable(std::shared_ptr<Function> func, std::string device);
+                virtual ~IE_Executable() {}
+                bool call(const std::vector<std::shared_ptr<runtime::Tensor>>& outputs,
+                          const std::vector<std::shared_ptr<runtime::Tensor>>& inputs) final;
+
+            private:
+                InferenceEngine::CNNNetwork network;
+                std::string device;
+            };
+
+            InferenceEngine::Blob::Ptr fill_blob(InferenceEngine::SizeVector shape,
+                                                 std::vector<float> data);
```

Comment:
Passing `data` by value seems expensive. Can you use `const std::vector<float>& data` instead?